### ISSUE TITLE
Overflow decode

### ISSUE THREAD

**Issue by Rikatemu** [Rikatemu/pyxelium#5]
When setting a very high base color, pixel values can overflow past 255. There's now a system that creates additional overflow pixels, but it cannot decode the image back to string correctly.

**Comment by ChaseSprigle** [comment]
I have been unable to recreate this. Using the latest commit was able to encode and then decode strings fine. What steps did you take to get an improperly decoded string?

**Comment by Rikatemu** [comment]
> I have been unable to recreate this. Using the latest commit was able to encode and then decode strings fine. What steps did you take to get an improperly decoded string?

@ChaseSprigle Sorry I didn't notice the comment. I've tried setting base color to 255, 255, 255 (everything will overflow) and it didn't work well when decoding :/
However it doesn't make much sense to make all pixels overflow as the resulting image is a garbage with useless pixels in it, so it might be better to put a value limit inside of the encode function, so that max value is something like 128 for each of the base color channels? What do you think?

**Comment by ChaseSprigle** [comment]
I just made a simple program that generates a random utf-8 string, encodes it, decodes it, and compares the decoded value to the original. Its appears to be working, so I'm not sure what is causing the decoding issues on your end.
In regards to reducing overflow pixels, we could certainly put a cap the max value of the base color, and that would reduce the overflow pixels. Depending on what direction you wanted to take the library, and if you are okay with making breaking changes, the overflow system could be removed entirely. If the base was added with overflow for encoding and subtracted with underflow for decoding, it would end up at the original value without needing to store a 255 somewhere. This would remove the need for overflow pixels and would make it so that 255 is stored in the other pixels less often.
